### DISCUSSION TITLE
Selection between multiple C++ libraries.

### DISCUSSION THREAD

**Discussion by antiqueverma** [toml-lang/toml#821] [Q&A] [answered]
Hi 
I am new to TOML. I was planning to use this file format with C++ for some basic applications.
Currently I am getting confused as in which C++ library will be best
Cktan or Marzer or ToruNiina. Can anyone please tell pros/cons (if any) of these three.

**Comment by marzer** [comment, marked as answer]
Obviously I have some skin in the game here, but it really depends on what your priorities are and what environment constraints you're limited to.
- What is the minimum version of the C++ standard you wish to target?
- Do you require TOML comment preservation?
- Do you prefer a more functional or imperative style of code?
- Do you prefer procedural C-style code, or modern C++-style code?
- Can/will you use exceptions?

All three have fairly different focuses so will be better or worse for you depending on your needs. Consider doing a quick scan of the documentation of each and see what feels like a good fit for you:
- [marzer/tomlplusplus](https://marzer.github.io/tomlplusplus/)
- [ToruNiina/toml11](https://github.com/ToruNiina/toml11/blob/master/README.md)
- [cktan/tomlcpp](https://github.com/cktan/tomlcpp/blob/main/README.md)
